### PR TITLE
[WIP] Add Softmax

### LINKED ISSUE #134: Implement Softmax function
## Feature Description
We would like to have an implementation of Softmax!

We already have some non linear function implemented through approximation, inluding [sigmoid](https://github.com/OpenMined/SyMPC/blob/b3f3ccd8c7452faa631e259448399f74e0c1abe6/src/sympc/approximations/sigmoid.py) but not yet Softmax, that would be super useful for Logistic Regression, which in turn is a building block for LSTM :)

### PR DESCRIPTION
**Pull request by Syzygianinfern0**
# TODO: Implement `torch.sum()`

## Description
Closes #134 

## Affected Dependencies
None. 

## How has this been tested?
- Added tests. 

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: src/sympc/tensor/static.py [hunk 1]
```diff
@@ -2,9 +2,6 @@
 
 Examples: torch.stack, torch.argmax
 """
-# future
-from __future__ import annotations
-
```
Comment (Syzygianinfern0) [inline src/sympc/tensor/static.py:7]: Something I left in by mistake in my earlier PR 😅

**Review by aanurraj — APPROVED** [review]
Niceeee

**Review by gmuraru — APPROVED** [review]
LGTM! Does this PR require more changes?

File: src/sympc/approximations/softmax.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    Returns:
+        MPCTensor: calculated MPCTensor
+    """
+    if dim is None:
+        dim = len(tensor.shape) - 1
+
+    # Single Element along dim
+    if tensor.shape[dim] == 1:
+        return 0 * tensor + 1  # Equivalent to torch.ones_like(tensor)
```
Comment (gmuraru) [inline src/sympc/approximations/softmax.py:27]: NITPICK: Use the PRZS rather than 0 * tensor

File: src/sympc/approximations/softmax.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    Returns:
+        MPCTensor: calculated MPCTensor
+    """
+    if dim is None:
+        dim = len(tensor.shape) - 1
+
+    # Single Element along dim
+    if tensor.shape[dim] == 1:
+        return 0 * tensor  # Equivalent to torch.zeros_like(tensor)
```
Comment (gmuraru) [inline src/sympc/approximations/softmax.py:56]: NITPICK: Use the PRZS rather than 0 * tensor

**Review by LaRiffle — APPROVED** [review]
Great job! 🕺

**Review by gmuraru — APPROVED** [review]
LGTM again!
